workspaces/ui-v2/src/store/endpoints/thunks.ts

```diff
@@ -63,10 +95,34 @@ export const endpointQueryResultsToJson = ({
     description: request.pathContributions.description || '',
     purpose: request.pathContributions.purpose || '',
     isRemoved: request.isRemoved,
+    query: request.query || null,
+    requestBodies: request.bodies.map((body) => ({
+      requestId: request.id,
+      contentType: body.contentType,
+      rootShapeId: body.rootShapeId,
+      pathId: request.pathId,
+      method: request.method,
+      description: request.requestContributions.description || '',
+    })),
+    responseBodies: request.responses
+      .flatMap((response) => {
+        return response.bodies.map((body) => {
+          return {
+            statusCode: response.statusCode,
```

Comment:
This `flatMap` -> `map` combo might indicate the same thing: that we should think of requests and responses separately. Haven't found the place where we actually use the fact that they're nested.